python-checks/src/main/java/org/sonar/python/checks/VerifiedSslTlsCertificateCheck.java

```diff
@@ -309,21 +307,38 @@ private static void urllibCheck(SubscriptionContext subscriptionContext) {
       .filter(Name.class::isInstance)
       .map(expr -> ((Name) expr).symbol())
       .ifPresent(symb -> {
-        for (Usage u : symb.usages()) {
+        for (Usage u : selectRelevantModifyingUsages(symb.usages(), vulnTok.token.line())) {
           searchForVerifyModeOverride(u).ifPresent(vulnTok::overrideBy);
         }
         if (vulnTok.isVulnerable) {
-          if (vulnTok.isInvisibleDefaultPreset) {
-            subscriptionContext.addIssue(
-              vulnTok.token,
-              "Certificate verification is disabled by default, verify_mode should be updated.");
-          } else {
-            subscriptionContext.addIssue(vulnTok.token, "Disabling certificate verification is dangerous.");
-          }
+          subscriptionContext.addIssue(vulnTok.token, MESSAGE);
         }
       }));
   }
 
+  /** Finds the next higher line where a binding usage occurs. */
+  private static int findNextAssignmentLine(List<Usage> usages, int firstAssignmentLine) {
```

Comment:
Remember that you don't have any guarantee about order of execution in usages list. They are sorted purely on syntax.

If you want to take into account order of execution, you should rely on CFG. However I don't think it's worth it for this check and for the FN mentioned in unit tests.
In my opinion, we should accept this FN when there are multiple assignements.

Reply:
After discussion with andrey, this is useful to mitigate FP on peach: https://peach.sonarsource.com/issues?languages=py&open=AXHKAcvKnkgMtVoYvDQs&resolved=false&rules=python%3AS4830
So we decided to accept the solution based on syntax for the time being.

Reply:
Indeed, the usages aren't even in the same order as they would occur in any syntax tree traversal - the assignments all come first, the "other".subscriptions all come later. Added sorting to selected relevant usages, just in case.